frontend/src/components/search/SearchOpportunityStatus.tsx

```diff
@@ -1,49 +1,82 @@
+import React, { useEffect, useState } from "react";
+
 import { Checkbox } from "@trussworks/react-uswds";
-import React from "react";
+import { useDebouncedCallback } from "use-debounce";
+import { useSearchParamUpdater } from "../../hooks/useSearchParamUpdater";
+
+interface StatusOption {
+  id: string;
+  label: string;
+  value: string;
+}
+
+interface SearchOpportunityStatusProps {
+  formRef: React.RefObject<HTMLFormElement>;
+}
+
+const statusOptions: StatusOption[] = [
+  { id: "status-forecasted", label: "Forecasted", value: "forecasted" },
+  { id: "status-posted", label: "Posted", value: "posted" },
+  { id: "status-closed", label: "Closed", value: "closed" },
+  { id: "status-archived", label: "Archived", value: "archived" },
+];
+
+const SearchOpportunityStatus: React.FC<SearchOpportunityStatusProps> = ({
+  formRef,
+}) => {
+  const [mounted, setMounted] = useState(false);
+  const { updateMultipleParam } = useSearchParamUpdater();
+  const [selectedStatuses, setSelectedStatuses] = useState<Set<string>>(
+    new Set(),
+  );
+
+  const handleCheck = (statusValue: string, isChecked: boolean) => {
+    setSelectedStatuses((prevSelectedStatuses) => {
+      const updatedStatuses = new Set(prevSelectedStatuses);
+      isChecked
+        ? updatedStatuses.add(statusValue)
+        : updatedStatuses.delete(statusValue);
+      return updatedStatuses;
+    });
+  };
+
+  // Wait half a second before updating query params
+  // and submitting the form
+  const debouncedUpdate = useDebouncedCallback(() => {
+    const key = "status";
+    updateMultipleParam(selectedStatuses, key);
+    formRef?.current?.requestSubmit();
+  }, 500);
```

Comment:
We might wait longer. I expect it can be more than a half second between clicks if someone is changing multiple checkboxes. Waiting a full second might be too long. Regardless, we'll definitely usability test this. What I'm wondering is whether it might be better to test a too-long debounce than a too-short one. @crystabelrangel, do you have any thoughts on how long this should be?

Reply:
Let's try 1 second